cedar-dafny/difftest/main.dfy

```diff
@@ -368,6 +374,19 @@ module difftest.main {
     response := responseToProdJson(ansAndErrors.0, ansAndErrors.1);
   }
 
+  const intTypeFromProdJson := objDeserializer2Fields(
+    "can_be_any", getJsonBool,
+    // Note: `bounds_opt` is optional in production, but it is omitted only for
+    // gradual bounds checking, which we don't support, so we can treat it as
+    // required. That doesn't give a great error message if the field is
+    // missing, but it doesn't seem worth the trouble to fix that.
+    "bounds_opt", objDeserializer2Fields("min", getJsonI64, "max", getJsonI64, (min, max) => Ok((min, max))),
```

Comment:
Wouldn't it make more sense to have one optional field "bounds" that is either `null` or `{min: ..., max: ...}`? What does it mean if both "can_be_any" and "bounds_opt" are set?

(Caveat: I haven't reviewed the Rust implementation, so apologies if my question is answered there.)